app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -11,12 +11,23 @@ def connect(options = {})
 
   def verify_credentials(_auth_type = nil, options = {})
     require 'azure-armrest'
-    connect(options)
-  rescue Azure::Armrest::UnauthorizedException
-    raise MiqException::MiqHostError, _("Incorrect credentials - check your Azure Client ID and Client Key")
+    conf = connect(options)
+
+    unless subscription.blank?
+      vms = ::Azure::Armrest::VirtualMachineService.new(conf)
+      subscriptions = vms.subscriptions.map(&:subscription_id)
+      unless subscriptions.include?(subscription)
+        raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Subscription ID")
+      end
+    end
+  rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
+    raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Client ID and Client Key")
   rescue StandardError => err
+    raise err if err.is_a?(MiqException::MiqInvalidCredentialsError)
```

Comment:
We want to differentiate between invalid credentials and "everything else". I don't think your way would do that, would it?